moto/core/responses.py

```diff
@@ -230,6 +234,24 @@ def __init__(self, service_name: Optional[str] = None):
     def dispatch(cls, *args: Any, **kwargs: Any) -> Any:  # type: ignore[misc]
         return cls()._dispatch(*args, **kwargs)
 
+    @classmethod
+    def method_dispatch(  # type: ignore[misc]
+        cls, to_call: Callable[[ResponseShape, Any, str, Any], TYPE_RESPONSE]
+    ) -> Callable[[Any, str, Any], TYPE_RESPONSE]:
+        """
+        Takes a given unbound function (part of a Response class) and executes it for a new instance of this
+        response class.
+        Can be used wherever we want to specify different methods for dispatching in urls.py
+        :param to_call: Unbound method residing in this Response class
+        :return: A wrapper executing the given method on a new instance of this class
+        """
+
+        @functools.wraps(to_call)  # type: ignore
```

Comment:
This is currently needed to pass the test `test_core.test_server.test_domain_dispatched_with_service` (with modifications). It not really wraps the method completely, so this should be discussed.